tests/benchmark/bm_utils.h

```diff
@@ -1,55 +1,192 @@
+#pragma once
+
 #include <benchmark/benchmark.h>
 #include <random>
 #include <unistd.h>
+#include <istream>
 #include "VecSim/vec_sim.h"
 #include "VecSim/vec_sim_interface.h"
 #include "VecSim/query_results.h"
 #include "VecSim/utils/arr_cpp.h"
 #include "VecSim/algorithms/brute_force/brute_force.h"
 #include "VecSim/algorithms/hnsw/hnsw_single.h"
+#include "bm_common.h"
 
-class BM_VecSimBasics : public benchmark::Fixture {
+class BM_VecSimGeneral : public benchmark::Fixture {
 public:
-    static VecSimIndex *bf_index;
-    static HNSWIndex<float, float> *hnsw_index;
+    static size_t block_size;
+
+protected:
     static size_t dim;
     static size_t n_vectors;
-    static size_t M;
-    static size_t EF_C;
-    static size_t block_size;
 
     static const char *test_vectors_file;
-    static std::vector<std::vector<float>> *queries;
     static size_t n_queries;
 
-    // We use this class as a singleton for every test case, so we won't hold several indices (to
-    // reduce memory consumption).
     static size_t ref_count;
 
-    BM_VecSimBasics();
-
-    static void Initialize();
-    static void RunTopK_HNSW(benchmark::State &st, size_t ef, size_t iter, size_t k,
-                             size_t &correct, VecSimIndex *hnsw_index_, VecSimIndex *bf_index_);
-    virtual ~BM_VecSimBasics();
+    static std::vector<VecSimIndex *> indices;
```

Comment:
we are saving the indices in an array of VecSimIndex so we can do generic tests (like AddVector and delete vector) instead of separate tests for hnsw and bf